src/edu/washington/escience/myria/api/encoding/IDBControllerEncoding.java

```diff
@@ -34,8 +36,13 @@
 
   @Override
   public IDBController construct(final ConstructArgs args) {
-    return new IDBController(argSelfIdbId, realEosControllerOperatorId, realEosControllerWorkerId, null, null, null,
-        argState.construct(), MoreObjects.firstNonNull(sync, Boolean.FALSE));
+    IDBController controller =
+        new IDBController(argSelfIdbId, realEosControllerOperatorId, realEosControllerWorkerId, null, null, null,
+            argState.construct(), MoreObjects.firstNonNull(sync, Boolean.FALSE));
+    if (relationKey != null) {
+      controller.setStoreRelationKey(relationKey);
```

Comment:
Why can't `relationKey` be supplied in the constructor if it isn't set afterward? Having setters that can be invoked after the constructor returns is less safe. If you feel the constructor args are getting out of control, consider using a builder.